kolibri/plugins/learn/assets/src/views/content-card-group-carousel/index.vue

```diff
@@ -87,6 +77,9 @@
       contents: {
         type: Array,
         required: true,
+        default() {
```

Comment:
Added to prevent rendering errors when not passed in any content. Was failing some tests, removes some responsibility from parent component

Reply:
It seems weird to have both `required: true` and a `default`. If there's a default, it means it's not required.

Reply:
I got that sense too, but the carousel would try and render content cards (and router links, which were what was giving the error that broke any tests with empty `contents`)when `contents` was null. I think `v-for` goes for it even if the value isn't iterable. Should this be expected behaviour?

Reply:
> I got that sense too, but the carousel would try and render content cards ... when contents was null. I think v-for goes for it even if the value isn't iterable. Should this be expected behaviour?

Yes, that's expected. Failing prop validation doesn't prevent the component from trying to render, it only provides warnings to the developer in vue's dev tools.

Your two options here are:

1. figure out why the parent or tests are passing `null` in as a value for `contents` and stop that from happening
2. keep the default value as `[]` and remove `required: true`

(1) seems more correct, but (2) is acceptable as a quick hack to just get things working and move forward